src/main/java/im/status/keycard/KeycardApplet.java

```diff
@@ -1405,25 +1511,19 @@ private void exportKey(APDU apdu) {
       len = publicKey.getW(apduBuffer, off);
       apduBuffer[(short) (off - 1)] = (byte) len;
       off += len;
-    } else if (publicOnly) {
+    } else {
       apduBuffer[off++] = TLV_PUB_KEY;
       off++;
       len = secp256k1.derivePublicKey(derivationOutput, (short) 0, apduBuffer, off);
       apduBuffer[(short) (off - 1)] = (byte) len;
       off += len;
     }
-
-    if (!publicOnly) {
-      apduBuffer[off++] = TLV_PRIV_KEY;
+    
+    if (withChaincode) {
+      apduBuffer[off++] = TLV_CHAIN_CODE;
       off++;
-
-      if (!derive || makeCurrent) {
-        len = privateKey.getS(apduBuffer, off);
-      } else {
-        Util.arrayCopyNonAtomic(derivationOutput, (short) 0, apduBuffer, off, Crypto.KEY_SECRET_SIZE);
-        len = Crypto.KEY_SECRET_SIZE;
-      }
-
+      Util.arrayCopyNonAtomic(chainCode, (short) 0, apduBuffer, off, CHAIN_CODE_SIZE);
+      len = CHAIN_CODE_SIZE;
```

Comment:
Why assign `len` here if it's just getting re-assigned 5 lines later? Reads kind of confusing; would prefer

```
apduBuffer[(short) (off - 1)] = (byte) CHAIN_CODE_SIZE;
 off += CHAIN_CODE_SIZE;
```

so that it's super duper explicit what we are doing

Reply:
Good point. I prefer the way you wrote it too. I will fix

Reply:
And now I remember why I did it this way. This entire function repeatedly copies data into the payload using the same 3 steps:
```
len = X; // Determine length to copy
Util.arrayCopyNonAtomic(off, len, ...); // Copy that length
off += len; // Update the offset
```

The paradigm makes sense (every offset update is programmatically dependent on the copy length), so I really wanted to keep with this paradigm. Also, consistency is friend 😄 

That said, if you really want me to change it, I can

Reply:
Nah it's fine